addons/web/static/src/core/select_menu/select_menu.js

```diff
@@ -69,12 +69,17 @@ export class SelectMenu extends Component {
         required: { type: Boolean, optional: true },
         searchable: { type: Boolean, optional: true },
         autoSort: { type: Boolean, optional: true },
+        placeholder: { type: String, optional: true },
         searchPlaceholder: { type: String, optional: true },
         value: { optional: true },
         multiSelect: { type: Boolean, optional: true },
         onInput: { type: Function, optional: true },
         onSelect: { type: Function, optional: true },
         slots: { type: Object, optional: true },
+        element: { type: HTMLElement, optional: true },
```

Comment:
At first sight, the props element and choiceFetchFunction seem weird

Reply:
why do we need to prepare choices with choiceFetchFunction here instead of doing it in parent widget?

Reply:
The reason why we added ```choiceFetchFunction``` is to Fetch the choices from server on the fly, we have some cases in website where we were using select2 lib and select2 was supporting data source to fetch data from server when user type something in select2 box, so have the same behavior we have added ```choiceFetchFunction```

Reply:
Hello, only the `placeholder` and `disabled` props are useful IMO.
The other props can be done with a wrapper component:
- choiceFetchFunction -> onInput prop
- onCreate -> bottomArea slot exists for this kind of usecase
- element -> element.value should be changed in an `onSelect` callback
```js
class SelectMenuWrapper extends Component {
  static components = { SelectMenu };
  static template = xml`
    <input t-att-value="state.value"/>
    <SelectMenu choices="state.choices" onInput.bind="loadChoices" onSelect.bind="onSelect">
      <t t-set-slot="bottomArea" t-slot-scope="select">
        <button class="btn btn-sm btn-primary m-1" t-on-click="() => this.onCreateOption(select.data.searchValue)">Create Option</button>
      </t>
    </SelectMenu>
  `;
  setup() {
    this.state = useState({ value: "", choices: [] });
    onWillStart(async () => this.loadChoices(""));
  }
  loadChoices(request) {
    const response = await fetch(`/load/choices?request=${request}`);
    this.state.choices = await response.json();
  }
  onSelect(value) {
    this.state.value = value;
  }
}
```

Reply:
@mcm-odoo OK Thanks for the suggestion, we will work on it soon and will update you guys once it is done.